MaxText/layers/moe.py

```diff
@@ -239,6 +264,13 @@ def generate_kernels(self, num_experts, emb_dim, mlp_dim):
 
   def get_topk(self, gate_logits, pre_bias_logits):
     """get topk. shape of top_k_weights & top_k_indices: (batch, sequence, num_experts_per_tok)"""
+    if self.config.use_random_routing:
+      # For debug and test purpose:
+      # Ideally we should split a new key for each layer, but for simplicity re-use the key to mimic the random routing
+      rng = jax.random.PRNGKey(1234)
```

Comment:
To get a real random routing, you need to change (split) the key at each generation.

Reply:
Yeah, I made a comment that `Ideally we should split a new key for each layer, but for simplicity re-use the key to mimic the random routing`. Otherwise, we need to pass it the from previous layer into next layer for split. I think this is for test and debug purpose, probably it's ok?

Reply:
you likely won't get nice distribution with reusing the key. We had a few bugs for repeated decode that producing the same token when key was reused.

Looks like you are splitting the key in the unit test.

Reply:
you can try something like this:

if self.config.use_random_routing:
      rng = self.make_rng('random_routing') 
      top_k_weights, top_k_indices = random_routing(rng, gate_logits, self.num_experts_per_tok)

Reply:
Good catch for the test!

You mean have a helper function to generate a key? Let me have a try. But will it be related to `random_routing` always?